tests/tablenavigation.js

```diff
@@ -0,0 +1,2284 @@
+							[ '10', `<horizontalLine></horizontalLine><paragraph>word[] ${ text }</paragraph>`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+
+						editor.editing.view.document.fire( 'keydown', upArrowDomEvtDataStub );
+
+						sinon.assert.calledOnce( upArrowDomEvtDataStub.preventDefault );
+						sinon.assert.calledOnce( upArrowDomEvtDataStub.stopPropagation );
+
+						assertEqualMarkup( getModelData( model ), modelTable( [
+							[ '00', '01', '02' ],
+							[ '10', `[<horizontalLine></horizontalLine>]<paragraph>word ${ text }</paragraph>`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+					} );
+
+					it( 'should move the caret to the end of cell content if the caret is in the last line of text', () => {
+						setModelData( model, modelTable( [
+							[ '00', '01', '02' ],
+							[ '10', `<paragraph>${ text } word []word</paragraph><horizontalLine></horizontalLine>`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+
+						editor.editing.view.document.fire( 'keydown', downArrowDomEvtDataStub );
+
+						sinon.assert.calledOnce( downArrowDomEvtDataStub.preventDefault );
+						sinon.assert.calledOnce( downArrowDomEvtDataStub.stopPropagation );
+
+						assertEqualMarkup( getModelData( model ), modelTable( [
+							[ '00', '01', '02' ],
+							[ '10', `<paragraph>${ text } word word</paragraph>[<horizontalLine></horizontalLine>]`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+					} );
+
+					it( 'should not move the caret to the end of cell content if widget is selected in middle of a cell content', () => {
+						const paragraph = `<paragraph>${ text }</paragraph>`;
+						const hr = '<horizontalLine></horizontalLine>';
+
+						setModelData( model, modelTable( [
+							[ '00', '01', '02' ],
+							[ '10', `${ paragraph }[${ hr }]${ paragraph }${ hr }`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+
+						editor.editing.view.document.fire( 'keydown', downArrowDomEvtDataStub );
+
+						assertEqualMarkup( getModelData( model ), modelTable( [
+							[ '00', '01', '02' ],
+							[ '10', `${ paragraph }${ hr }<paragraph>[]${ text }</paragraph>${ hr }`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+					} );
+
+					it( 'should not move the caret to the end of cell content if widget is next to the selection', () => {
+						const paragraph = `<paragraph>${ text }</paragraph>`;
+						const hr = '<horizontalLine></horizontalLine>';
+
+						setModelData( model, modelTable( [
+							[ '00', '01', '02' ],
+							[ '10', `${ paragraph }[]${ hr }`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+
+						editor.editing.view.document.fire( 'keydown', downArrowDomEvtDataStub );
+
+						assertEqualMarkup( getModelData( model ), modelTable( [
+							[ '00', '01', '02' ],
+							[ '10', `${ paragraph }[${ hr }]`, '12' ],
+							[ '20', '21', '22' ]
+						] ) );
+					} );
+				} );
+
+				describe( 'contains image widget with caption and selection inside the caption', () => {
```

Comment:
What contains that image? TCs suggest this is about a cell where the initial selection is but it is not clear from the description. It could as well describe a place the selection should be anchored to after the navigation.

Reply:
Table cell contains image.